lib/vrl/stdlib/src/find.rs

```diff
@@ -111,7 +111,7 @@ impl Expression for FindFn {
     }
 
     fn type_def(&self, _: &state::Compiler) -> TypeDef {
-        TypeDef::new().infallible().integer()
+        TypeDef::integer().infallible()
```

Comment:
I think all the `.infallible()` suffixes in this file should be redundant, right? Might be able to grep for more in this PR.

```suggestion
        TypeDef::integer()
```

Reply:
Correct, but I'm going to leave this as-is, there are many places this pattern is used, we can change it, but I consider it outside the scope of this PR.